dbmigrator/commands/mark.py

```diff
@@ -19,17 +19,27 @@ def cli_command(cursor, migrations_directory='', migration_timestamp='',
     if completed is None:
         raise Exception('-t, -f or -d must be supplied.')
 
-    migrations = utils.get_migrations(
-        migrations_directory, import_modules=False)
-    for version, _ in migrations:
-        if version == migration_timestamp:
-            break
-    else:
+    migrations = {version: migration
+                  for version, _, migration in utils.get_migrations(
+                      migrations_directory, import_modules=True)}
+    migration = migrations.get(migration_timestamp)
+    if migration is None:
         migrated_versions = list(utils.get_schema_versions(cursor))
         if migration_timestamp not in migrated_versions:
-            logger.warning(
+            logger.error(
                 'Migration {} not found'.format(migration_timestamp))
+            return
```

Comment:
Hmm, since the case is that the user-supplied value is wrong, yes, I think it should stop and return 1 (failure). I can't think of a non-error case for that behavior.